### PR TITLE
Search map

### PR DESCRIPTION
**Pull request by eshurakov**
### Details
In this PR we're adding a search map to the editor. 

Ref: #626

### Screenshots

Screenshot 2021-02-02 at 20 25 52
Screenshot 2021-02-02 at 20 26 02

### Test (Check for big and small notes)

1. Search something

- [ ] Search map is shown on the side of the editor
- [ ] Search map is shown only when editor is not focused

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated in a41d37b with:
> Add search map to the editor

### CODE REVIEW

File: Simplenote/NoteEditorViewController+Swift.swift [hunk 1]
```diff
@@ -941,16 +941,77 @@ extension NoteEditorViewController {
...
         guard !noteEditor.isFirstResponder, !keywords.isEmpty else {
-            noteEditor.highlightedRanges = []
-            return
+            return []
         }
 
         let slice = noteEditor.attributedString().string.contentSlice(matching: keywords)
-        noteEditor.highlightedRanges = slice?.nsMatches ?? []
+        return slice?.nsMatches ?? []
+    }
```
Comment (jleandroperez) [inline Simplenote/NoteEditorViewController+Swift.swift:953]: I really really like this change!!

File: Simplenote/NoteEditorViewController+Swift.swift [hunk 2]
```diff
@@ -941,16 +941,77 @@ extension NoteEditorViewController {
...
+            searchMapView.topAnchor.constraint(equalTo: scrollView.topAnchor, constant: SplitItemMetrics.editorScrollerTopInset)
+        ])
+
+        self.searchMapView = searchMapView
+    }
+
+    /// Returns position relative to the total text container height.
+    /// Position value is from 0 to 1
+    ///
+    private func searchBarPositions(with searchRanges: [NSRange]) -> [CGFloat] {
```
Comment (jleandroperez) [inline Simplenote/NoteEditorViewController+Swift.swift]: WDYT if `searchBarPositions` and `textContainerHeightForSearchMap` were moved over SPTextView?

We could rename this API as follows (and shrink a bit the VC?)

```
searchBarPositions` > `relativeLocationsForText(in ranges: [NSRange]) -> [CGFloat]
```

**Review by jleandroperez — APPROVED** [review]
@eshurakov Rock solid!!! 

Added just one cosmetic comment, feel free to disregard!!

:shipit: sir!!

**Review by SylvesterWilmott — APPROVED** [review]

File: Simplenote/NoteEditorViewController+Swift.swift (hunk 2 above)
Comment (eshurakov) [reply Simplenote/NoteEditorViewController+Swift.swift]: Good idea, let me try :-)

**Comment by eshurakov** [comment]
Thank you!

**Comment by pachlava** [comment]
[testing] fix/feature verified on MBP 16" Intel | macOS 11.2 | Simplenote 2.8 (11218)

**Pull request merged**